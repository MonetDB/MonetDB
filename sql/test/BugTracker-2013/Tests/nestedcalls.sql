CREATE TABLE tnc (name VARCHAR(10));
INSERT INTO tnc VALUES ('test');

CREATE FUNCTION bottom()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from tnc;
END;

CREATE FUNCTION leaf1()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from bottom()
	UNION  select name from bottom()
	UNION  select name from bottom()
	UNION  select name from bottom();
END;

CREATE FUNCTION leaf2()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from leaf1()
	UNION  select name from leaf1()
	UNION  select name from leaf1()
	UNION  select name from leaf1();
END;

CREATE FUNCTION leaf3()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from leaf2()
	UNION  select name from leaf2()
	UNION  select name from leaf2()
	UNION  select name from leaf2();
END;

SELECT count(*) FROM bottom();
SELECT count(*) FROM leaf1();
SELECT count(*) FROM leaf2();
SELECT count(*) FROM leaf3();

DROP FUNCTION leaf3;
DROP FUNCTION leaf2;
DROP FUNCTION leaf1;
DROP FUNCTION bottom;
DROP TABLE tnc;
