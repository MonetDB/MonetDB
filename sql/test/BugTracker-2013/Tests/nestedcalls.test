statement ok
CREATE TABLE tnc (name VARCHAR(10))

statement ok
INSERT INTO tnc VALUES ('test')

statement ok
CREATE FUNCTION bottom()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from tnc;
END

statement ok
CREATE FUNCTION leaf1()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from bottom()
	UNION  select name from bottom()
	UNION  select name from bottom()
	UNION  select name from bottom();
END

statement ok
CREATE FUNCTION leaf2()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from leaf1()
	UNION  select name from leaf1()
	UNION  select name from leaf1()
	UNION  select name from leaf1();
END

statement ok
CREATE FUNCTION leaf3()
RETURNS TABLE (name string)
BEGIN
	RETURN select name from leaf2()
	UNION  select name from leaf2()
	UNION  select name from leaf2()
	UNION  select name from leaf2();
END

query I rowsort
SELECT count(*) FROM bottom()
----
1

query I rowsort
SELECT count(*) FROM leaf1()
----
1

query I rowsort
SELECT count(*) FROM leaf2()
----
1

query I rowsort
SELECT count(*) FROM leaf3()
----
1

statement ok
DROP FUNCTION leaf3

statement ok
DROP FUNCTION leaf2

statement ok
DROP FUNCTION leaf1

statement ok
DROP FUNCTION bottom

statement ok
DROP TABLE tnc

