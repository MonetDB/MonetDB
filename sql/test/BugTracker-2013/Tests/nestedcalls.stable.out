stdout of test 'nestedcalls` in directory 'sql/test/BugTracker-2013` itself:


# 09:11:23 >  
# 09:11:23 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=37677" "--set" "mapi_usock=/var/tmp/mtest-1158/.s.monetdb.37677" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/Feb2013//Linux/var/MonetDB/mTests_sql_test_BugTracker-2013" "--set" "mal_listing=0"
# 09:11:23 >  

# MonetDB 5 server v11.15.12
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2013', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Found 15.591 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2015 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:37677/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-1158/.s.monetdb.37677
# MonetDB/GIS module loaded
# MonetDB/JAQL module loaded
# MonetDB/SQL module loaded

# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_history.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_compress.sql
# loading sql script: 18_dictionary.sql
# loading sql script: 19_cluster.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 24_zorder.sql
# loading sql script: 25_debug.sql
# loading sql script: 39_analytics.sql
# loading sql script: 40_geom.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_udf.sql
# loading sql script: 99_system.sql

# 09:11:23 >  
# 09:11:23 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-1158" "--port=37677"
# 09:11:23 >  

#CREATE TABLE tnc (name VARCHAR(10));
#INSERT INTO tnc VALUES ('test');
[ 1	]
#CREATE FUNCTION bottom()
#RETURNS TABLE (name string)
#BEGIN
#	RETURN select name from tables;
#END;
#CREATE FUNCTION leaf1()
#RETURNS TABLE (name string)
#BEGIN
#	RETURN select name from bottom()
#	UNION  select name from bottom()
#	UNION  select name from bottom()
#	UNION  select name from bottom();
#END;
#CREATE FUNCTION leaf2()
#RETURNS TABLE (name string)
#BEGIN
#	RETURN select name from leaf1()
#	UNION  select name from leaf1()
#	UNION  select name from leaf1()
#	UNION  select name from leaf1();
#END;
#CREATE FUNCTION leaf3()
#RETURNS TABLE (name string)
#BEGIN
#	RETURN select name from leaf2()
#	UNION  select name from leaf2()
#	UNION  select name from leaf2()
#	UNION  select name from leaf2();
#END;
#SELECT count(*) FROM bottom();
% .%2 # table_name
% %2 # name
% bigint # type
% 1 # length
[ 1	]
#SELECT count(*) FROM leaf1();
% .%2 # table_name
% %2 # name
% bigint # type
% 1 # length
[ 1	]
#SELECT count(*) FROM leaf2();
% .%2 # table_name
% %2 # name
% bigint # type
% 1 # length
[ 1	]
#SELECT count(*) FROM leaf3();
% .%2 # table_name
% %2 # name
% bigint # type
% 1 # length
[ 1	]
#DROP FUNCTION leaf3;
#DROP FUNCTION leaf2;
#DROP FUNCTION leaf1;
#DROP FUNCTION bottom;

# 09:11:24 >  
# 09:11:24 >  "Done."
# 09:11:24 >  

